tests/test_temporal.py

```diff
@@ -1104,6 +1104,114 @@ async def test_toolset_without_id():
         TemporalAgent(Agent(model=model, name='test_agent', toolsets=[FunctionToolset()]))
 
 
+# --- DynamicToolset / @agent.toolset tests ---
+
+
+def get_dynamic_weather(location: str) -> str:
+    """Get the weather for a location."""
+    return f'The weather in {location} is sunny.'
+
+
+dynamic_toolset_agent = Agent(TestModel(), name='dynamic_toolset_agent')
+
+
+@dynamic_toolset_agent.toolset
+def my_dynamic_toolset(ctx: RunContext[None]) -> FunctionToolset[None]:
+    return FunctionToolset(tools=[get_dynamic_weather], id='dynamic_weather')
```

Comment:
Let's use a value from `deps` inside the function (when we inline it), since the point of `DynamicToolset` is to build it dynamically. We can do that in the MCP server example as well, if that seems more appropriate